api/test/unit/views/media_views_test.py

```diff
@@ -241,3 +243,30 @@ def test_thumb_full_size(api_client, media_type, media_factory, requests):
         assert (
             entry in requests.sent_requests[1].request.url
         ), f"{entry} not found in prepared request url: {requests.sent_requests[0].request.url}"
+
+
+@pytest.fixture
+def redis(monkeypatch) -> FakeRedis:
+    fake_redis = FakeRedis()
+    monkeypatch.setattr("catalog.api.views.media_views.cache", fake_redis)
+
+    yield fake_redis
+
+    fake_redis.client().close()
+
+
+@pytest.mark.parametrize("count", [1, 3])
+def test_thumb_timeout(redis, count: int):
+    with patch("requests.get", side_effect=ReadTimeout()):
+        for idx in range(count):
+            try:
+                MediaViewSet._thumbnail_proxy_comm(
+                    "test", params={"url": f"https://example.com/image/{idx}"}
+                )
+            except UpstreamThumbnailException:
+                # Intentionally raised exception, do nothing.
+                pass
```

Comment:
Should the test use `pytest.raises` I wonder? If we expect the exception to be raised then it'd be nice if the unit test encapsulated that aspect.

Reply:
Resolved in 6b24482.